### ISSUE TITLE
Compression Error

### ISSUE THREAD

**Issue by g-l-i-t-c-h-o-r-s-e** [ruarai/Trigrad#2]
Hey there, compiled Trigrad on Windows 7 32bit.
When I run TrigradTesting.exe after I had to manually make an tests/input folder with a file in it, it crashes after generating mesh_a.png and points.png with this error message:
```
Trigrad
memory init completed in 1151
memory buffer init completed in 189
memory init 2 completed in 1
Unhandled Exception: Cloo.MemoryObjectAllocationFailureComputeException: OpenCL
error code detected: MemoryObjectAllocationFailure.
   at Cloo.ComputeException.ThrowOnError(ComputeErrorCode errorCode)
   at Cloo.ComputeCommandQueue.Username(ComputeKernel kernel, Int64[] globalWorkO
ffset, Int64[] globalWorkSize, Int64[] localWorkSize, ICollection`1 events)
   at Trigrad.GPUT.Calculate(List`1 calculations) in C:\Users\Username\Downloads\
Trigrad-master\Trigrad-master\Trigrad\GPUT.cs:line 96
   at Trigrad.GPUT.CalculateMesh(List`1 mesh) in C:\Users\Username\Downloads\Trig
rad-master\Trigrad-master\Trigrad\GPUT.cs:line 27
   at Trigrad.TrigradOptimiser.OptimiseMesh(TrigradCompressed compressionData, P
ixelMap original, TrigradOptions options) in C:\Users\Username\Downloads\Trigrad-
master\Trigrad-master\Trigrad\TrigradOptimiser.cs:line 31
   at TrigradTesting.Program.Main(String[] args) in C:\Users\Username\Downloads\T
rigrad-master\Trigrad-master\TrigradTesting\Program.cs:line 49
```
pls halp? I built Trigrad via msbuild (VS 2017).

**Comment by g-l-i-t-c-h-o-r-s-e** [comment]
Update: This is happening because the input image resolution was 2048x1536, changing the input source to a smaller image successfully compressed the image. Is there a way to make input image resolution larger without crashing?

**Comment by ruarai** [comment]
Yeah, if I remember correctly the GPU acceleration doesn't account for VRAM.

Looking at the code I haven't got a clue how I'd fix it. I remember doing something for it a looong time ago. But it's been four years since I've worked on this, so I don't know how I'd fix it at this point.

Here's a UI I found lying around that may or may not help you: https://www.dropbox.com/s/90slz0bthkb83zc/triart.zip?dl=0

Did you also email me?
